src/shared/state/assets.ts

```diff
@@ -0,0 +1,32 @@
+import { ViewService } from '@penumbra-zone/protobuf';
+import { getDenomMetadata } from '@penumbra-zone/getters/assets-response';
+import { ChainRegistryClient } from '@penumbra-labs/registry';
+import { penumbra } from '@/shared/penumbra';
+import { connectionStore } from '@/shared/state/connection';
+import { useQuery } from '@tanstack/react-query';
+import { useEnv } from '@/fetchers/env';
+
+export const useAssets = () => {
+  const { data: env, isLoading } = useEnv();
+  const chainId = !isLoading ? env?.PENUMBRA_CHAIN_ID : process.env['PENUMBRA_CHAIN_ID'];
+
+  const registryAssets = useQuery({
+    queryKey: ['registry-assets'],
+    queryFn: () => {
+      const registryClient = new ChainRegistryClient();
+      const registry = registryClient.bundled.get(chainId ?? '');
+      return registry.getAllAssets();
+    },
+  });
+
+  const accountAssets = useQuery({
+    queryKey: ['assets'],
+    enabled: connectionStore.connected,
+    queryFn: async () => {
+      const responses = await Array.fromAsync(penumbra.service(ViewService).assets({}));
+      return responses.map(getDenomMetadata);
+    },
+  });
+
+  return connectionStore.connected ? accountAssets : registryAssets;
```

Comment:
issue: we should be propagating errors also from the querys this hook consumes. That means we have to combine the states. If you look into `useRegistry()` there is an example of this. Would look like:

```ts
  return connectionStore.connected
    ? accountAssets
    : { ...registryAssets, error: registryAssets.error ?? registryErr };
```

Reply:
thought: I think, we don't need to propagate the loading state as if `enabled: false` I think `registryAssets.isLoading` is true (?). Not sure should check that. If the loading state isn't propagated via the enabled field of registryAssets, we should need to do the same as: https://github.com/penumbra-zone/dex-explorer/blob/df94a11f1d0d8e8fcdddeaa238471da719676696/src/fetchers/registry.ts#L29. 

However, registryAssets is in a loading state while `enabled: false`, we can do as I suggested above and simplify the pattern like my example in useRegistry()